src/ng/directive/ngOptions.js

```diff
@@ -686,53 +650,32 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
 
             if (!group) {
 
-              // We have not already created this group
-              groupElement = addOrReuseElement(selectElement[0],
-                                               currentElement,
-                                               'optgroup',
-                                               optGroupTemplate);
-              // Move to the next element
-              currentElement = groupElement.nextSibling;
+              groupElement = optGroupTemplate.cloneNode(false);
+              listFragment.appendChild(groupElement);
 
               // Update the label on the group element
               groupElement.label = option.group;
 
               // Store it for use later
               group = groupMap[option.group] = {
-                groupElement: groupElement,
-                currentOptionElement: groupElement.firstChild
+                groupElement: groupElement
```

Comment:
Minor: Since now this object only holds `groupElement`, we can avoid creating a new object and j7st store the `groupElement` directly.